File: src/main/java/fr/epicanard/mapsaver/utils/Messenger.java
package fr.epicanard.mapsaver.utils;

import net.md_5.bungee.api.chat.TextComponent;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class Messenger {
    public static String prefix;

    public static void setPrefix(final String prefix) {
        Messenger.prefix = toColor(prefix);
    }

    public static void sendMessage(final CommandSender sender, final String message) {
        sender.sendMessage(prefix + toColor(message));
    }

    public static void sendMessage(final CommandSender sender, final TextComponent message) {
        sender.spigot().sendMessage(message);
    }

    public static void sendMessage(final Player player, final String message) {
        player.sendMessage(toColor(message));
    }

    public static String toColor(final String message) {
        return ChatColor.translateAlternateColorCodes('&', message);
    }

}

File: src/main/scala/fr/epicanard/mapsaver/Messenger.scala
package fr.epicanard.mapsaver

import fr.epicanard.mapsaver.resources.language.{ErrorMessages, Help, Language}
import org.bukkit.command.CommandSender

case class Messenger(prefix: String, language: Language) {
  def sendError(sender: CommandSender)(getMessage: ErrorMessages => String): Unit =
    send(sender, s"&c${getMessage(language.errorMessages)}")

  def sendHelp(sender: CommandSender)(getMessage: Help => String): Unit =
    send(sender, getMessage(language.help))

  private def send(sender: CommandSender, message: String): Unit =
    sender.sendMessage(toColor(s"$prefix$message"))

  private def toColor(message: String): String =
    message.replaceAll("&", "§")
}
